File: 485-max-consecutive-ones/max-consecutive-ones.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int count = 0, max = 0;

        for(int i = 0; i < nums.size(); i++){
            if(nums[i] == 1) {
                count++;
            } else if(nums [i] == 0) {
                max = (max < count) ? count : max;
                count = 0;
            }
        }

        return (max < count) ? count : max;;
    }
};

File: 485-max-consecutive-ones/max-consecutive-ones.java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
      int max = 0;
      int count = 0;

      for(int num : nums) {
          if (num == 1) {
              count++;
          } else {
              max = count > max ? count : max;
              count = 0;
          }
      }
      
      max = count > max ? count : max;
      return max;  
    }
}